MycroftAI/mycroft-core
# Use XDG Base directories for settings, cache and runtime data

## Description
This makes Mycroft use the [XDG Base Directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
Data is now saved to:
* settings to `$XDG_CONFIG_DIR/mycroft`, which most of the time equals to `~/.config/mycroft`
* data to `$XDG_DATA_DIRS/mycroft`, which most of the time equals to `~/.local/share/mycroft`
* cache to `$XDG_CACHE_HOME/mycroft`, which most of the time equals to `~/.cache`

Since this changes the default location of pretty much everything, _please test it well_!

Goal is to support and read from the old `~/.mycroft` location if it already exists, but save to the XDG Base Directories and read from it on new installations.

Together with #2559 this should be full XDG Base Directory compliance.

## How to test
First of all, run this PR with an existing installation, and check if everything is loaded from the usual locations. Then shut it down and restart to make sure it has saved to the new locations.
Then, completely remove any trace of Mycroft settings from your system. Easiest would be to fully deinstall (remove `/opt/mycroft`), remove `~/.mycroft`, and the XDG Base directories mentioned above.
Now reinstall Mycroft (with this PR applied) and start and connect it. Check if `~/.mycroft` is _not_ created anymore and `~/.config/mycroft`, `~/.local/share/mycroft` and `~/.cache/mycroft` _are_ with the correct files and directories in them.

## Contributor license agreement signed?
CLA [x]
